pkg/services/serviceaccounts/proxy/service.go

```diff
@@ -0,0 +1,100 @@
+package proxy
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/apikey"
+	"github.com/grafana/grafana/pkg/services/extsvcauth/extsvcaccounts"
+	"github.com/grafana/grafana/pkg/services/serviceaccounts"
+	"github.com/grafana/grafana/pkg/services/serviceaccounts/manager"
+)
+
+// ServiceAccountsProxy is a proxy for the serviceaccounts.Service interface
+// that is used to add validations to service accounts and protects external
+// service accounts from being modified by users.
+
+type ServiceAccountsProxy struct {
+	log            log.Logger
+	proxiedService serviceaccounts.Service
+}
+
+func ProvideServiceAccountsProxy(
+	proxiedService *manager.ServiceAccountsService,
+) (*ServiceAccountsProxy, error) {
+	s := &ServiceAccountsProxy{
+		log:            log.New("serviceaccounts.proxy"),
+		proxiedService: proxiedService,
+	}
+	return s, nil
+}
+
+var _ serviceaccounts.Service = (*ServiceAccountsProxy)(nil)
+
+func (s *ServiceAccountsProxy) CreateServiceAccount(ctx context.Context, orgID int64, saForm *serviceaccounts.CreateServiceAccountForm) (*serviceaccounts.ServiceAccountDTO, error) {
+	if isExternalServiceAccount(saForm.Name) {
```

Comment:
Since the check is on the name not the generated login, we shouldn't check against `sa-extsvc-` but against `extsvc-`

Reply:
Changed as requested